src/textual/widget.py

```diff
@@ -993,9 +993,21 @@ def on_blur(self, event: events.Blur) -> None:
 
     def on_descendant_focus(self, event: events.DescendantFocus) -> None:
         self.descendant_has_focus = True
+        if "focus-within" in self.pseudo_classes:
```

Comment:
Maybe this `if` block can be factorised to a private method, since it seems to be exactly the same in the 2 methods?

But we might also want to wait for the hypothetic 3rd occurrence of it to appear before doing so :slightly_smiling_face: 
> The Rule of Three
> 
>     The first time you do something, you just do it. The second time you do
>     something similar, you wince at the duplication, but you do the duplicate
>     thing anyway. The third time you do something similar, you refactor.

Reply:
Good point. If there is a third, we probably want a `refresh_children`.